stock_picking_force_assign/tests/test_stock_picking_force_assign.py

```diff
@@ -12,16 +12,14 @@ def setUp(self, *args, **kwargs):
             'name': 'Buy',
             'rule_ids': [(0, 0, {
                 'name': 'Buy',
-                'action': 'buy',
+                'action': 'pull',
                 'picking_type_id': self.env.ref('stock.picking_type_in').id,
                 'location_id': self.env.ref('stock.stock_location_stock').id,
             })]
         })
         route_mto = self.env.ref('stock.route_warehouse0_mto')
-        self.product = self.env['product.product'].create({
-            'name': 'stockable',
-            'type': 'product',
-        })
+        self.product = self.product = self.env.ref('product.product_delivery_01')
+        self.location = self.env.ref('stock.stock_location_stock')
```

Comment:
If you must copy the location in a variable, why don't you do that before line 17 and use it there as well?